containers/roles/client-rpm/files/build-client-rpm/tasks/main.yml

```diff
@@ -0,0 +1,34 @@
+---
+- name: 'Make build directory'
+  file:
+    state: directory
+    path: "/root/build"
+
+- name: 'Read ca.crt'
+  set_fact:
+    ca_crt: "{{ lookup('file', '/etc/pki/katello/certs/katello-default-ca.crt') }}"
```

Comment:
Does this need to be added at build time? Is this something that will ever need to be replaced by a user? They would have to rebuild the container right?

Reply:
This container is a bit of Ansible-inception so bear with me. The built image contains a role that performs this task. Then when the container is executed it looks up this data from the mounted secret, generates an RPM from it and pushes that RPM to the pub/ PVC. The only time this needs to be re-run is if the CA crt is regenerated which is a super rare event. I don't think overall this is the cleanest solution but I am attempting to mimic the behavior of existing installs. I do wonder if the Openshift part should be a Job rather than a deployment? I am less familiar with Jobs but saw you use them for the data import.

Reply:
Ah this file (`main.yml`) is being built into the image itself.

A Job is a `run a container until success` sort of operation, so if this does not need to exist after running and will only run once then a Job sounds like a good fit.

Reply:
Updated to Job type and moved into a jobs file